sympy/core/mod.py

```diff
@@ -97,11 +97,14 @@ def doit(p, q):
         if p.is_Add:
             args = []
             for i in p.args:
-                a = cls(i, q)
-                if a.count(cls) > i.count(cls):
-                    args.append(i)
+                if (i/q).is_integer:
+                    pass
```

Comment:
This could probably be `continue`. Then the next `else` (and added indentation) would not be needed.